src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,136 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_userUnderMinAge_notOk() {
+        User user = new User("user", "123456", 17);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAgeNegative_notOk() {
+        User user = new User("user", "123456", -1);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_passwordUnderMinLength_notOk() {
+        User user = new User("user", "12345", 18);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_alreadyExistedLogin_notOk() {
+        User loginAlreadyExist = new User("LoginAlreadyExist", "123456", 19);
+        StorageDaoImpl storageDao = new StorageDaoImpl();
+        storageDao.add(loginAlreadyExist);
```

Comment:
It's better to use Storage directly(what if storageDao.add() method does not work correctly?)